stl/inc/chrono

```diff
@@ -2309,15 +2339,110 @@ namespace chrono {
+        unique_ptr<__std_tzdb_current_zone_info, decltype(&__std_tzdb_delete_current_zone)> _Info{
+            __std_tzdb_get_current_zone(), &__std_tzdb_delete_current_zone};
+        if (_Info == nullptr) {
+            _Xbad_alloc();
+        } else if (_Info->_Err == __std_tzdb_error::_Win_error) {
+            _XGetLastError();
+        } else if (_Info->_Err == __std_tzdb_error::_Icu_error) {
+            _Xruntime_error("FIXME: what generic message should I use here?");
+        }
+
+        return {_Info->_Tz_name};
+    }
+
+    template <class _Ty>
+    _NODISCARD const _Ty* _Locate_zone_impl(const std::vector<_Ty, _Crt_allocator<_Ty>>& _Vec, string_view _Name) {
+        const auto _Result = find_if(_Vec.begin(), _Vec.end(), [&](auto& _Tz) { return _Tz.name() == _Name; });
+        return _Result != _Vec.end() ? &(*_Result) : nullptr;
+    }
+
     struct tzdb {
+        vector<time_zone, _Crt_allocator<time_zone>> time_zones;
+        vector<time_zone_link, _Crt_allocator<time_zone_link>> links;
         vector<leap_second, _Crt_allocator<leap_second>> leap_seconds;
         bool _All_ls_positive;
+
+        _NODISCARD const time_zone* locate_zone(string_view _Tz_name) const {
+            auto _Tz = _Locate_zone_impl(time_zones, _Tz_name);
+            if (_Tz == nullptr) {
+                const auto _Link = _Locate_zone_impl(links, _Tz_name);
+                if (_Link != nullptr) {
+                    _Tz = _Locate_zone_impl(time_zones, _Link->target());
+                }
+            }
+
+            return _Tz;
+        }
+
+        _NODISCARD const time_zone* current_zone() const {
+            return locate_zone(_Xtzdb_generate_current_zone());
+        }
     };
 
+    _NODISCARD inline pair<decltype(tzdb::time_zones), decltype(tzdb::links)> _Xtzdb_generate_time_zones() {
+        unique_ptr<__std_tzdb_time_zones_info, decltype(&__std_tzdb_delete_time_zones)> _Info{
+            __std_tzdb_get_time_zones(), &__std_tzdb_delete_time_zones};
+        if (_Info == nullptr) {
+            _Xbad_alloc();
+        } else if (_Info->_Err == __std_tzdb_error::_Win_error) {
+            _XGetLastError();
+        } else if (_Info->_Err == __std_tzdb_error::_Icu_error) {
+            _Xruntime_error("FIXME: what generic message should I use here?");
+        }
+
+        decltype(tzdb::time_zones) _Time_zones;
+        decltype(tzdb::links) _Time_zone_links;
+        for (size_t _Idx = 0; _Idx < _Info->_Num_time_zones; ++_Idx) {
+            const std::string_view _Name{_Info->_Names[_Idx], strlen(_Info->_Names[_Idx])};
+            if (_Info->_Links[_Idx] == nullptr) {
+                _Time_zones.push_back(time_zone{_Name});
+            } else {
+                const std::string_view _Target{_Info->_Links[_Idx], strlen(_Info->_Links[_Idx])};
+                _Time_zone_links.push_back(time_zone_link{_Name, _Target});
+            }
+        }
+
+        return {_Time_zones, _Time_zone_links};

```

Comment:
I think these should be `_STD move()`d, we aren't going to get that implicitly when constructing a `pair`.